### PR TITLE
Signup: add AB to test domain results popover

### PR DESCRIPTION
**Pull request by michaeldcain**
As part of #7134, we'd like to test the effectiveness of the "Included in WordPress.com Premium" popover in the domain results listing by running a variation that hides the popover.

**Hypothesis from #7303:**

> The idea here is that by removing all of the extraneous information from this step, we may see a higher percentage of users who actually complete the domain step and continue to the next part — picking a plan.

**`showPopover` (Default) Variant:**
showPopover screenshot

**`hidePopover` Variant:**
hidePopover screenshot

Test live: https://calypso.live/start/?branch=add/signup-domains-popover-ab

### CODE REVIEW

**Comment by coreh** [comment]
LGTM! :thumbsup:

**Comment by bisko** [comment]
I think 841d3ae makes the code a little bit more cleaner. 

cc @michaeldcain @coreh for comments.

File: client/components/domains/domain-product-price/index.jsx [hunk 1]
```diff
@@ -93,8 +102,8 @@ const DomainProductPrice = React.createClass( {
 } );
 
 export default connect(
-	state => ( { domainsWithPlansOnly: getCurrentUser( state ) ?
-		currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY ) :
-		true
+	state => ( { domainsWithPlansOnly: getCurrentUser( state )
+		? currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+		: true
```
Comment (mcsf) [inline client/components/domains/domain-product-price/index.jsx]: Note that the logic `a ? b : true` can be reduced to `! a || b`.

**Comment by coreh** [comment]
LGTM 👍

File: client/components/domains/domain-product-price/index.jsx (hunk 1 above)
Comment (coreh) [inline client/components/domains/domain-product-price/index.jsx]: | `a` | `b` | `a ? b : true` | `! a ||  b` |
| --- | --- | --- | --- |
| `true` | `true` |  `true` | `true` |
| `true` | `false` |  `false` | `false` |
| `false` | `true` |  `true` | `true` |
| `false` | `false` |  `false` | `true` 🚫 |

Comment (sirbrillig) [inline client/components/domains/domain-product-price/index.jsx]: While I like reduction to avoid ternaries, I think readability trumps terseness. IMO keep the ternary for clarity.

Comment (mcsf) [inline client/components/domains/domain-product-price/index.jsx]: Funny, it reads pretty clearly to me, but it might be that my brain is broken due to long-term abuse.

The way I intuitively think of `! a || …` is "if there is no `a`, well that's enough [of a condition]; otherwise, we need _something more_. In the case at hand, that means:

`! getCurrentUser( state ) || currentUserHasFlag( state, … )`

which reads:
- no user? that's fine
- otherwise, make sure they have the right flag

This stands in contrast with the common pattern:

`getCurrentUser( state ) && currentUserHasFlag( state, … )`

which reads:
- no user? that's no good, _we needs it_
- also, we need the flags

To put it another way, for me the pattern isn't so much `! a || b` as just `! a ||`. It's a lenient test, or an easy early-bailing mechanism, compared with the strong `a &&` pattern.

Comment (mcsf) [inline client/components/domains/domain-product-price/index.jsx]: (Obviously, if it still doesn't read well, I'm not gonna defend the binary expr 😄, but I wanted to share my interpretation.)
